src/client/dfs/dfs.c

```diff
@@ -4681,6 +4716,12 @@ dfs_osetattr(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf, int flags)
 		recx[i].rx_idx = MTIME_IDX;
 		recx[i].rx_nr = sizeof(time_t);
 		i++;
+
+		d_iov_set(&sg_iovs[i], &obj_hlc, sizeof(time_t));
```

Comment:
Since ATTR_UID has the same "issue", I think it makes sense to fix both together (perhaps as a separate PR).
Automating a test that uses different versions is tricky, but probably an entire area of testing that needs work.